src/Microsoft.Azure.WebJobs.Host/Bindings/BindingDataProvider.cs

```diff
@@ -96,15 +96,37 @@ public static BindingDataProvider FromType(Type type)
             }
 
             Dictionary<string, Type> contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+            Dictionary<string, Type> declaringTypes = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
```

Comment:
We need to track the declaring type for inspection.  I considered changing the `contract`  set to `Dictionary<string, PropertyInfo>` and then using a `ToDictionary` call to transform for the return set, but I believe using an additional dictionary will be more efficient.